### PR TITLE
Pascal/mar 473 define implement api for automatic index creation

### PR DESCRIPTION
**Pull request by Pascal-Delange**
Preliminary for the back-end work required for https://github.com/checkmarble/marble-frontend/pull/367

Naming is open for discussion: I went for /commit but I'm not set on it

### CODE REVIEW

File: repositories/transaction.go [hunk 1]
```diff
@@ -1,37 +0,0 @@
-package repositories
```
Comment (Pascal-Delange) [inline repositories/transaction.go:1]: forgot to completely remove that yesterday

**Comment by Pascal-Delange** [comment]
Tested & working locally

File: usecases/scenario_iterations_usecase.go [hunk 2]
```diff
@@ -252,3 +318,34 @@ func replaceTriggerOrRule(scenarioAndIteration scenarios.ScenarioAndIteration,
...
+		models.GetScenarioIterationFilters{ScenarioId: &scenarioId})
+	if err != nil {
+		return 0, err
+	}
+
+	for _, scenarioIteration := range scenarioIterations {
+		if scenarioIteration.Id == iterationId && scenarioIteration.Version != nil {
+			return *scenarioIteration.Version, nil
+		}
+	}
```
Comment (balzdur) [inline usecases/scenario_iterations_usecase.go:340]: Not required, or should return an error (not the current version).

It's due to the copy paste: the old method was doing two things in one, either retrieve the existing version number (= reactivate) or compute the next version to commit the draft (=activate)

**Review by balzdur — APPROVED** [review]

File: usecases/scenario_iterations_usecase.go (hunk 2 above)
Comment (Pascal-Delange) [reply usecases/scenario_iterations_usecase.go:340]: ah yes, my bad

**Comment by Pascal-Delange** [comment]
big last commit: add tests

**Pull request merged**